pkg/collector/corechecks/ebpf/probe/oom_kill.go

```diff
@@ -40,7 +40,7 @@ type OOMKillProbe struct {
 }
 
 func NewOOMKillProbe(cfg *ebpf.Config) (*OOMKillProbe, error) {
-	compiledOutput, err := runtime.OomKill.Compile(cfg, nil)
+	compiledOutput, err := runtime.OomKill.Compile(cfg, []string{"-Wno-error=int-conversion"})
```

Comment:
Just curious, do you know what lines are causing these errors?

Reply:
These really should not be ignored if at all possible.

Reply:
For the oom kill check:
https://github.com/DataDog/datadog-agent/blob/729162b115e91331040de46450626cdd28ef1ece/pkg/collector/corechecks/ebpf/c/runtime/oom-kill-kern.c#L34

For the tcp queue length check:
https://github.com/DataDog/datadog-agent/blob/729162b115e91331040de46450626cdd28ef1ece/pkg/collector/corechecks/ebpf/c/runtime/tcp-queue-length-kern.c#L106 and https://github.com/DataDog/datadog-agent/blob/729162b115e91331040de46450626cdd28ef1ece/pkg/collector/corechecks/ebpf/c/runtime/tcp-queue-length-kern.c#L125

If it's an issue, I can remove this flag & just cast the return values of `PT_REGS_PARM1` to solve this error, that's fine too